smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/SubscriberMediator.java

```diff
@@ -127,20 +127,36 @@ public void onComplete() {
     }
 
     private void processMethodReturningVoid() {
-        this.subscriber = ReactiveStreams.<Message<?>> builder()
-                .flatMapCompletionStage(managePreProcessingAck())
-                .map(message -> {
-                    invoke(message.getPayload());
-                    return message;
-                })
-                .flatMapCompletionStage(x -> {
-                    if (configuration.getAcknowledgment() == Acknowledgment.Strategy.POST_PROCESSING) {
-                        return getAckOrCompletion(x);
-                    } else {
-                        return CompletableFuture.completedFuture(x);
-                    }
-                })
-                .ignore();
+        if (configuration.isBlocking()) {
+            this.subscriber = ReactiveStreams.<Message<?>> builder()
+                    .flatMapCompletionStage(managePreProcessingAck())
+                    .flatMapCompletionStage(message -> ((Uni<?>) invokeBlocking(message.getPayload()))
+                            .subscribeAsCompletionStage()
+                            .thenApply(x -> message))
+                    .flatMapCompletionStage(x -> {
+                        if (configuration.getAcknowledgment() == Acknowledgment.Strategy.POST_PROCESSING) {
```

Comment:
We may want to refactor and extract this code as it's now duplicated a few times. It can be done in another PR.